stl/inc/flat_map

```diff
@@ -722,8 +722,9 @@ public:
     }
 
     _NODISCARD_FRIEND auto operator<=>(const _Derived& _Left, const _Derived& _Right) {
-        return _STD lexicographical_compare_three_way(
-            _Left.cbegin(), _Left.cend(), _Right.cbegin(), _Right.cend(), _Synth_three_way{});
+        return _STD lexicographical_compare_three_way(_STD _Get_unwrapped(_Left.cbegin()),
+            _STD _Get_unwrapped(_Left.cend()), _STD _Get_unwrapped(_Right.cbegin()), _STD _Get_unwrapped(_Right.cend()),
+            _Synth_three_way{});

```

Comment:
Oops, the unwrapping here doesn't seem to work.
Should we add unwrapping mechanism to the specific iterators or manually unwrap the underlying iterators here?

Reply:
We should probably teach `flat_meow` iterators to unwrap properly, so all algorithms can take advantage of it.